TheMET/TheMET/Services/FeaturedArtService.swift

```diff
@@ -127,34 +125,43 @@ class FeaturedArtService {
         self.featuredArt = .loading
         self.metAPI.objects { [weak self] objectResponce in
             guard let objectResponce = objectResponce,
-            let randomId = objectResponce.objectIDs.randomElement() else {
+                  let randomId = self?.getRandomID(objectToDelete: nil, objectResponce: objectResponce) else {
                 self?.featuredArt = .failed
                 self?.isFeaturedArtLoading = false
                 return
             }
-            self?.metAPI.object(id: randomId) {[weak self] object in
-                guard let object = object else {
+            self?.tryRandomId(randomID: randomId, objectResponce: objectResponce)
+        }
+    }
+    
+    private func getRandomID (objectToDelete: ArtID?, objectResponce: ObjectsResponse) -> ArtID? {
+        objectResponce.objectIDs.removeAll { $0 == objectToDelete }
+        guard let randomId = objectResponce.objectIDs.randomElement() else {return nil}
+        return randomId
+    }
+    
+    private func tryRandomId(randomID: ArtID, objectResponce: ObjectsResponse) {
+        self.metAPI.object(id: randomID) { [weak self] object in
+            guard let object = object else {
+                self?.featuredArt = .failed
+                self?.isFeaturedArtLoading = false
+                return
+            }
+            guard let imageString = object.primaryImage,
+                let _ = URL(string: imageString) else {
```

Comment:
use `URL(string: imageString) != nil` instead of `let _ = URL(string: imageString)` as it is more explicit check